### DISCUSSION TITLE
[Question] Lots of path variables missing when using SSHClient login remote

### DISCUSSION THREAD

**Discussion by Roy-Kid** [ParallelSSH/parallel-ssh#332] [Q&A] [answered]
Hi, I have a question that. When I use SSHClient to connect to remote HPC, I find a lot of commands can not be found, such as 'qstat', `module load`. I inspect the `$PATH` and find some PATH variables which the administrator installs are missing. But those I write in `.bashrc` stay. I wonder why and how to solve this problem. 

Sorry for asking this silly question and hope you can help me. Thanks!

**Comment by pkittenis** [comment, marked as answer]
Hi,

That's to do with how bash behaves in remote execute mode (what happens when `run_command` is called). In this case only `.bashrc` is read and not `.bash_profile`. Presumably some paths are set only in `.bash_profile` and so do not get set on `run_command`.

Either use an interactive shell which uses a login shell which does read `.bash_profile`, or set paths in bashrc or any other method.

[See bash details here](https://unix.stackexchange.com/questions/332531/why-does-remote-bash-source-bash-profile-instead-of-bashrc).

See also `man bash` and `man ssh`.

**Reply by Roy-Kid** [reply]
Thanks for your help and sorry for replying late. Your attached link is so important to help me understand this question. I will rewrite my code this weekend. I want to start a session with login shell mode.
